beacon-chain/core/blocks/block_operations.go

```diff
@@ -830,6 +867,9 @@ func ProcessDeposits(
 	var err error
 	deposits := body.Deposits
 	for _, deposit := range deposits {
+		if deposit == nil || deposit.Data == nil {
+			continue
```

Comment:
again it should fail over here

Reply:
i changed to continue after a discussion with @terencechain i also thought this is a cause for failure. no nil deposit should be in a list of deposits. changing back to failure

Reply:
I see. I think i agree fail is more sensible here. I chose `cont` at first to be more align with the spec logic. Switching to fail is good with me

Reply:
thanks